profile/src/main/java/com/scottlogic/deg/profile/reader/atomic/ConstraintValueValidator.java

```diff
@@ -0,0 +1,188 @@
+package com.scottlogic.deg.profile.reader.atomic;
+
+import com.scottlogic.deg.common.ValidationException;
+import com.scottlogic.deg.common.profile.Field;
+import com.scottlogic.deg.common.profile.constraintdetail.ParsedDateGranularity;
+import com.scottlogic.deg.common.profile.constraintdetail.ParsedGranularity;
+import com.scottlogic.deg.common.util.Defaults;
+import com.scottlogic.deg.common.util.NumberUtils;
+import com.scottlogic.deg.generator.fieldspecs.whitelist.DistributedSet;
+import com.scottlogic.deg.profile.dto.AtomicConstraintType;
+import com.scottlogic.deg.profile.reader.InvalidProfileException;
+
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.util.regex.Pattern;
+
+import static com.scottlogic.deg.profile.dto.AtomicConstraintType.IS_NULL;
+
+public class ConstraintValueValidator {
+
+    public static void validate(String field, AtomicConstraintType type, Object value){
+        try {
+            validateConstraintValue(type, value);
+        } catch (IllegalArgumentException | ValidationException e){
+            throw new InvalidProfileException(String.format("Field [%s]: %s", field, e.getMessage()));
+        }
+    }
+
+    private static void validateConstraintValue(AtomicConstraintType type, Object value){
+
+        if (type == IS_NULL){
+            validateNull(value);
+            return;
+        }
+
+        validateNotNull(value);
+
+        switch (type) {
+            case IS_EQUAL_TO_CONSTANT:
+                validateAny(type, value);
+                break;
+            case IS_IN_SET:
+                validateSet(type, value);
+                break;
+            case IS_OF_TYPE:
+                validateTypes(value);
+                break;
+
+            case MATCHES_REGEX:
+            case CONTAINS_REGEX:
+                validatePattern(value);
+                break;
+
+            case HAS_LENGTH:
+            case IS_STRING_SHORTER_THAN:
+            case IS_STRING_LONGER_THAN:
+                validateStringLengthInt(type, value);
+                break;
+
+            case IS_GREATER_THAN_CONSTANT:
+            case IS_GREATER_THAN_OR_EQUAL_TO_CONSTANT:
+            case IS_LESS_THAN_CONSTANT:
+            case IS_LESS_THAN_OR_EQUAL_TO_CONSTANT:
+                validateNumber(type, value);
+                break;
+
+            case IS_AFTER_CONSTANT_DATE_TIME:
+            case IS_AFTER_OR_EQUAL_TO_CONSTANT_DATE_TIME:
+            case IS_BEFORE_CONSTANT_DATE_TIME:
+            case IS_BEFORE_OR_EQUAL_TO_CONSTANT_DATE_TIME:
+                validateDateTime(value);
+                break;
+
+            case IS_GRANULAR_TO:
+                validateGranularity(value);
```

Comment:
if using switches need to have a default line which throws exception so we get early alert if add new type and forget to add to all the switches

Reply:
I'll do this change on the next branch, as the constraints mess up the required cases